pkg/connector/group.go

```diff
@@ -162,83 +184,175 @@ func (g *groupBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken
 		}
 	}
 
-	// Get user permissions
-	userPermissions, err := listGlobalUserPermissions(ctx, g.client)
-	if err != nil {
-		switch {
-		case errors.As(err, &bitbucketErr):
-			if bitbucketErr.ErrorCode != http.StatusUnauthorized {
-				return nil, "", nil, fmt.Errorf("%s", bitbucketErr.Error())
-			}
-		default:
+	groupName := resource.Id.Resource
+	switch bag.ResourceTypeID() {
+	case resourceTypeRepository.Id:
+		repos, nextPageToken, err := g.client.ListRepos(ctx, client.PageOptions{
+			PerPage: ITEMSPERPAGE,
+			Page:    pageToken,
+		})
+		if err != nil {
 			return nil, "", nil, err
 		}
-	}
 
-	// Get group permissions
-	groupPermissions, err := listGlobalGroupPermissions(ctx, g.client)
-	if err != nil {
-		switch {
-		case errors.As(err, &bitbucketErr):
-			if bitbucketErr.ErrorCode != http.StatusUnauthorized {
-				return nil, "", nil, fmt.Errorf("%s", bitbucketErr.Error())
+		for _, repo := range repos {
+			projectKey := repo.Project.Key
+			repositorySlug := repo.Slug
+			groupRepositoryPermissions, err := listGroupRepositoryPermissions(ctx, g.client, projectKey, repositorySlug)
+			if err != nil {
+				return nil, "", nil, err
 			}
-		default:
+
+			groupsPermissionsPos := slices.IndexFunc(groupRepositoryPermissions, func(c client.GroupsPermissions) bool {
+				return c.Group.Name == groupName
+			})
+			if groupsPermissionsPos != NF {
```

Comment:
Nitpick: You could un-indent a lot of this code by continuing early:

```suggestion
			if groupsPermissionsPos == NF {
				continue
			}
```

Reply:
Fixed